### DISCUSSION TITLE
How to bind to a specific indexer, if there are multiple

### DISCUSSION THREAD

**Discussion by Flying--Dutchman** [AvaloniaUI/Avalonia#13512] [Q&A]
Hello, I'm trying to bind a DataGrid-Column to a specific indexer, using `new Binding("[0]")` which throws an `AmbiguousMatchException` during runtime, since the class that I'm targeting contains 2 indexers. One for `int` and one for `string`. 
The trick from [this post](https://stackoverflow.com/questions/5738585/how-can-a-wpf-binding-distinguish-between-an-indexer-property-and-a-list-element) did not work unfortunately.

**Comment by maxkatz6** [comment]
Can you create a feature request for `<TextBlock Text="{Binding Path=[(sys:Int32)0]}"/>` syntax? We already have casting support in compiled bindings, and it makes sense to support it in this scenario as well.

Right now you can create a converted as a workaround.

**Reply by Flying--Dutchman** [reply]
> Can you create a feature request for `<TextBlock Text="{Binding Path=[(sys:Int32)0]}"/>` syntax? We already have casting support in compiled bindings, and it makes sense to support it in this scenario as well.

> Right now you can create a converted as a workaround.

I'm not using XAML for this, but directly in the code behind. Do you have an example for me?

**Reply by timunie** [reply]
You can still add a converter for it. Just pass the IValueConverter to your Binding. in the converter, `value` would be your class with indexer and `ConverterParameter` the index to lookup.
